### ISSUE TITLE
Fun token and mintable fun token should refund when user's balance is 0

### ISSUE THREAD

**Issue by nearmax** [near/core-contracts#80]
As reported by a user:
> Should the burn function refund the sender if they burn all their balance? If I have X tokens as balance and burn them all, the env::storage_usage after the call is less. I think we should be refunded (since we initially paid for that storage when minting!). So there should be a self.refund_storage(initial_storage) above this line https://github.com/near/rainbow-bridge/blob/master/libs-rs/mintable-fungible-token/src/lib.rs#L479, along with a let initial_storage = env::storage_usage() in the first line of the burn function

This would require changes for both types of contracts in two different repos.

**Comment by evgenykuzyakov** [comment]
Should I move this to `rainbow-bridge`? Original fungible token doesn't have `burn` method.

**Comment by nearmax** [comment]
@evgenykuzyakov That's true, I forgot about that. Please move it to rainbow-bridge and assign to @ailisp . Also make it P1. Thank you!
